components/stack_exchange/stack_exchange.app.js

```diff
@@ -61,35 +156,51 @@ module.exports = {
         headers,
       };
     },
-    async *advancedSearch(baseParams) {
+    advancedSearch(searchParams) {
       const url = this._advancedSearchUrl();
-      const baseRequestConfig = this._makeRequestConfig();
+      return this.getItems(url, searchParams);
+    },
+    answersForQuestions(questionIds, searchParams) {
+      const url = this._answersForQuestionsUrl(questionIds);
+      return this.getItems(url, searchParams);
+    },
+    answersFromUsers(userIds, searchParams) {
+      const url = this._answersFromUsersUrl(userIds);
+      return this.getItems(url, searchParams);
+    },
+    async *getItems(url, baseParams) {
       let page = 1;
       let hasMore = false;
       do {
-        const params = {
-          ...baseParams,
-          page,
-        };
-        const requestConfig = {
-          ...baseRequestConfig,
-          params,
-        };
-        const { data } = await axios.get(url, requestConfig);
+        const data = await this.getItemsForPage(url, baseParams, page);
         if (data.items.length === 0) {
```

Comment:
Will `data.items` always be defined?

Reply:
Yes, for the URL's for which we're calling this method. If no items are present, the API returns an empty array.
I will add a check just in case so that we fail gracefull should something change.